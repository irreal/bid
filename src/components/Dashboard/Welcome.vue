<template>
  <div class="main-header__welcome">
    <div class="main-header__welcome-title text-light">
      Zdravo,
      <strong>{{ username }}</strong>
    </div>
    <div class="main-header__welcome-subtitle text-light"></div>
  </div>
</template>
<script>
import { mapGetters } from "vuex";
export default {
  name: "Welcome",
  computed: {
    ...mapGetters({ user: "getUser" }),
    username() {
      return this.user ? this.user.username : "neulogovani korisnik";
    }
  }
};
</script>
<style scoped lang="scss"></style>
