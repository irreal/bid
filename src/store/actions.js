export default {
  updateUser: ({ commit }, payload) => {
    commit("setUser", payload.currentUser);
  }
};
