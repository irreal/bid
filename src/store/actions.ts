import { ActionTree } from 'vuex';
import State from './state';

const actions: ActionTree<State, State> = {
    //   updateColor({ commit }: ActionContext<State, State>, newColor: string): void {
    //     commit(COLOR, newColor);
    //   },
    };

export default actions;
