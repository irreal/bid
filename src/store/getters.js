export default {
  getUserStatus: state => !!state.user,
  getUser: state => state.user
};
