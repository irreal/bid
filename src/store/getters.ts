import { GetterTree } from 'vuex';
import State from './state';
import Plan from '@/models/plan';

const getters: GetterTree<State, State> = {
    plans(state: State): Plan[] {
      return state.plans;
    },
    // colour(state: State): string {
    // 	return state.color;
    // },
  };

export default getters;
