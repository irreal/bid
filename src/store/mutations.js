export default {
  setUser: (state, currentUser) => {
    state.user = currentUser;
  }
};
