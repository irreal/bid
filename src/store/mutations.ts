import { MutationTree } from 'vuex';
import State from './state';

const mutations: MutationTree<State> = {
    removePlans(state: State) {
      state.plans = [];
    },
    // color(state: State, newColor: string): void {
    // 	state.color = newColor;
    // }
  };

export default mutations;
