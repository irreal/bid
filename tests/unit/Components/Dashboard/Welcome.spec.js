import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Welcome from "@/components/Dashboard/Welcome";
import Vuetify from "vuetify";
import Vue from "vue";
import initialState from "@/store/state";
import actions from "@/store/actions";
import getters from "@/store/getters";
import mutations from "@/store/mutations";
import userFixture from "../../fixtures/user";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

describe("Welcome component", () => {
  let state;

  const build = () => {
    const welcome = shallowMount(Welcome, {
      localVue,
      store: new Vuex.Store({ state, actions, getters, mutations })
    });

    return {
      welcome
    };
  };

  beforeEach(() => {
    state = { ...initialState };
  });
  it("renders the component", () => {
    // arrange
    const { welcome } = build();
    // assert
    expect(welcome.html()).toMatchSnapshot();
    expect(welcome.html()).toContain("neulogovani korisnik");
  });

  it("renders the component with a logged in user", () => {
    // arrange
    state.user = userFixture;
    const { welcome } = build();
    // assert
    expect(welcome.html()).toMatchSnapshot();
    expect(welcome.html()).toContain(userFixture.user_metadata.full_name);
  });
});
