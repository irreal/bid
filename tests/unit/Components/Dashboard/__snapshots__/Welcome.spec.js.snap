// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component renders the component 1`] = `<v-layout-stub tag="div" align-end="" py-3=""><span class="headline">Zdravo, <strong>neulogovani korisnik</strong></span></v-layout-stub>`;

exports[`Welcome component renders the component with a logged in user 1`] = `<v-layout-stub tag="div" align-end="" py-3=""><span class="headline">Zdravo, <strong>Miloš Spasojević</strong></span></v-layout-stub>`;
